cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/ZookeeperScaler.java

```diff
@@ -324,18 +327,13 @@ private ZKClientConfig getClientConfig()  {
      * @param scale     Number of nodes which the Zookeeper cluster should have
      * @return          Map with configuration
      */
-    /*test*/ static Map<String, String> generateConfig(int scale)   {
+    /*test*/ static Map<String, String> generateConfig(int scale, Function<Integer, String> zkNodeAddress)   {
         Map<String, String> servers = new HashMap<>(scale);
 
         for (int i = 0; i < scale; i++) {
-            // The Zookeeper server IDs starts with 1
+            // The Zookeeper server IDs starts with 1, but pod index starts from 0
             String key = String.format("server.%d", i + 1);
-
-            // The ports
-            int followerPort = (ZookeeperCluster.CLUSTERING_PORT * 10) + i;
-            int electionPort = (ZookeeperCluster.LEADER_ELECTION_PORT * 10) + i;
-            int clientPort = (ZookeeperCluster.CLIENT_PORT * 10) + i;
-            String value = String.format("127.0.0.1:%d:%d:participant;127.0.0.1:%d", followerPort, electionPort, clientPort);
+            String value = String.format("%s:%d:%d:participant;127.0.0.1:%d", zkNodeAddress.apply(i), ZookeeperCluster.CLUSTERING_PORT, ZookeeperCluster.LEADER_ELECTION_PORT, 2181);
```

Comment:
Do we want to use a constant and reference it for the `2181` here

Reply:
Done